### PR TITLE
Add support for Doctrine embeddable entities

### PR DESCRIPTION
**Pull request by mnugter**
Add support for Doctrine embeddable entities

### CODE REVIEW

**Comment by mnugter** [comment]
This time in 1 commit and hopefully with approval of Travis..

File: src/Helper/FieldManager.php [hunk 1]
```diff
@@ -239,15 +281,30 @@ protected function setRelation(string $entity, string $sourceEntity = null): str
...
+
+    /**
+     * Get the relation metadata for the provided entity.
+     *
+     * @param $sourceEntity
+     * @param $entity
+     *
+     * @return array
+     */
+    protected function getRelationMetaData($sourceEntity, $entity)
```
Comment (paknahad) [inline src/Helper/FieldManager.php]: please define the type of parameters & return value.

File: src/Helper/FieldManager.php [hunk 2]
```diff
@@ -239,15 +281,30 @@ protected function setRelation(string $entity, string $sourceEntity = null): str
...
+          'alias' => $alias,
         ];
 
         return $this->relations[$entity]['entityClass'];
     }
+
+    /**
+     * Get the relation metadata for the provided entity.
+     *
+     * @param $sourceEntity
```
Comment (paknahad) [inline src/Helper/FieldManager.php]: @param string $sourceEntity

File: src/Helper/FieldManager.php [hunk 3]
```diff
@@ -239,15 +281,30 @@ protected function setRelation(string $entity, string $sourceEntity = null): str
...
 
-        $associations = $this->entityManager->getClassMetadata($sourceEntity)->associationMappings;
+        $relationMetaData = $this->getRelationMetaData($sourceEntity, $entity);
 
         $this->relations[$entity] = [
-            'entity' => $associations[$entity]['fieldName'] ?? null,
-            'entityClass' => $associations[$entity]['targetEntity'] ?? $entity,
-            'sourceEntity' => $sourceEntity,
-            'alias' => $alias,
+          'entity' => $relationMetaData['fieldName'] ?? null,
```
Comment (paknahad) [inline src/Helper/FieldManager.php]: please keep code's integrity, use 4 space instead of 2 space for indenting.

Comment (mnugter) [reply src/Helper/FieldManager.php]: I fixed the coding standards issues.

File: src/Helper/FieldManager.php (hunk 2 above)
Comment (mnugter) [reply src/Helper/FieldManager.php]: I fixed the coding standards issues.

File: src/Helper/FieldManager.php (hunk 1 above)
Comment (mnugter) [reply src/Helper/FieldManager.php]: I fixed the coding standards issues.

**Comment by mnugter** [comment]
Added various typehints for string throughout the whole class.

**Pull request merged**